client/src/Components/relatedItems/relatedCarousel/relatedCard/cardComponents/comparison.jsx

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+function Comparison({ trigger, handlesComparePopup, data, displayProduct }) {
+  return (
+    trigger ?
+    <div className="comparison-popup"> {data.name} {displayProduct.name}
+      <button onClick={() => handlesComparePopup()}>X</button>
```

Comment:
If 'handlesComparePopup' does not take in any arguments, it's not necessary to create a function that calls handlesComparePopup.

Instead, try 'onClick={handlesComparePopup}